python/lsst/obs/base/makeRawVisitInfoViaObsInfo.py

```diff
@@ -0,0 +1,172 @@
+    VisitInfo information as possible from a messy FITS header without the
+    user needing to add a lot of error handling.
+
+    Parameters
+    ----------
+    log : `lsst.log.Log` or None
+        Logger to use for messages.
+        (None to use ``Log.getLogger("MakeRawVisitInfoViaObsInfo")``).
+    """
+
+    metadataTranslator = None
+    """Header translator to use to construct VisitInfo, defaulting to
+    automatic determination."""
+
+    def __init__(self, log=None):
+        if log is None:
+            log = Log.getLogger("MakeRawVisitInfoViaObsInfo")
+        self.log = log
+
+    def __call__(self, md, exposureId=None):
+        """Construct a VisitInfo and strip associated data from the metadata.
+
+        Parameters
+        ----------
+        md : `lsst.daf.base.PropertyList` or `lsst.daf.base.PropertySet`
+            Metadata to pull from.
+            Items that are used are stripped from the metadata.
+        exposureId : `int`, optional
+            Ignored.  Here for compatibility with `MakeRawVisitInfo`.
+
+        Returns
+        -------
+        visitInfo : `lsst.afw.image.VisitInfo`
+            `~lsst.afw.image.VisitInfo` derived from the header using
+            a `~lsst.obs.metadata.MetadataTranslator`.
+        """
+        argDict = dict()
+
+        obsInfo = ObservationInfo(md, translator_class=self.metadataTranslator)
+        print(obsInfo.__dict__)
+
+        # Strip all the cards out that were used
+        for c in obsInfo.cards_used:
+            del md[c]
+
+        # Map the translated information into a form suitable for VisitInfo
+        argDict["exposureTime"] = obsInfo.exposure_time
+        argDict["darkTime"] = obsInfo.dark_time
+        argDict["exposureId"] = obsInfo.detector_exposure_id
+
+        # VisitInfo uses the middle of the observation for the date
+        if obsInfo.datetime_begin is not None and obsInfo.datetime_end is not None:
+            tdelta = obsInfo.datetime_end - obsInfo.datetime_begin
+            middle = obsInfo.datetime_begin + 0.5*tdelta
+
+            # DateTime uses nanosecond resolution, regardless of the resolution
+            # of the original date
+            middle.precision = 9
+            # isot is ISO8601 format with "T" separating date and time and no
+            # time zone
+            argDict["date"] = DateTime(middle.tai.isot, DateTime.TAI)
+
+            # Derive earth rotation angle from UT1 (being out by a second is not
+            # a big deal given the uncertainty over exactly what part of the
+            # observation we are needing it for).
+            # ERFA needs a UT1 time split into two floats
+            # We ignore any problems with DUT1 not being defined for now.
+            try:
+                ut1time = middle.ut1
+            except iers.IERSRangeError:
+                ut1time = middle
+
+            era = erfa.era00(ut1time.jd1, ut1time.jd2)
+            argDict["era"] = era * radians
```

Comment:
This seems like the kind of calculation I'd have expected to happen inside obs_metadata.  Are you just trying to be compatible with VisitInfo's definition of a quantity, while disliking that definition enough to not want it in ObservationInfo?

Reply:
Earth rotation angle is entirely a property of the time of observation. It's not something that I felt should exist in ObservationInfo and it's not something that is really needed if you are using Astropy coordinates everywhere.  afw VisitInfo needs it because it's not got access to astropy. This calculation exists here entirely to be compatible with afw.